lib/match.js

```diff
@@ -120,6 +113,10 @@ function match(object, matcherOrValue) {
             break;
     }
 
+    if (isArrayType(object) && isArrayType(matcherOrValue)) {
```

Comment:
Instead of adding another `if` statement (I worked hard at removing those), you could list the array types in the switch statement ... it's not like they're going to change a lot :)

Reply:
Was thinking about the same, first. Good argument 👍